eng/pipelines/templates/jobs/archetype-go-release.yml

```diff
@@ -44,15 +44,16 @@ stages:
   - stage: Release
     variables:
       - template: /eng/pipelines/templates/variables/globals.yml
+      - template: /eng/pipelines/templates/variables/image.yml
     displayName: 'Release: ${{ parameters.ServiceDirectory }}'
     dependsOn: CheckRelease
     condition: and(succeeded(), eq(dependencies.CheckRelease.outputs['CheckReleaseJob.Verify.NeedToRelease'], 'true'))
     jobs:
       - deployment: ReleaseGate
         environment: package-publish
         pool:
-          name: azsdk-pool
-          image: ubuntu-24.04
+          name: $(LINUXPOOL)
```

Comment:
Unfortunately DevOps does not allow you to variables in deployment jobs so they have to be hard-coded.